satellite/metainfo/metainfo.go

```diff
@@ -1247,9 +1223,55 @@ func (endpoint *Endpoint) BeginSegment(ctx context.Context, req *pb.SegmentBegin
 		return nil, status.Errorf(codes.Unauthenticated, err.Error())
 	}
 
-	// TODO implement logic
+	// no need to validate streamID fields because it was validated during BeginObject
+
+	exceeded, limit, err := endpoint.projectUsage.ExceedsStorageUsage(ctx, keyInfo.ProjectID)
+	if err != nil {
+		endpoint.log.Error("retrieving project storage totals", zap.Error(err))
+	}
+	if exceeded {
+		endpoint.log.Sugar().Errorf("monthly project limits are %s of storage and bandwidth usage. This limit has been exceeded for storage for projectID %s",
```

Comment:
Why not use here a `log.Error` rather than the sugar version?

Reply:
It was copied from old implementation. I didn't want to touch too much and also it quite readable for me. I can change that if you feel its important.

Reply:
oh, I could not see in the diff